docs/guide/en/result.md

```diff
@@ -281,3 +281,23 @@ $result->getAttributeErrors('email');
 ```
 
 [Using keys containing separator / shortcut]: #using-keys-containing-separator--shortcut
+
+## Predefined result
+
+Sometimes you can have information about validation errors before validation process. In this case use predefined result
+and pass it to validator via parameter `ValidationContext::PARAMETER_PREDEFINED_RESULT` of validation context:
+
+```php
+use Yiisoft\Validator\Result;
+use Yiisoft\Validator\Validator;
+use Yiisoft\Validator\ValidationContext;
+
+$predefinedResult = new Result();
+$predefinedResult->addError('Invalid value.', valuePath: ['key']);
+
+$context = new ValidationContext([
+    ValidationContext::PARAMETER_PREDEFINED_RESULT => $predefinedResult,
+])
+
+$result = (new Validator)->validate($model, context: $context);
```

Comment:
I considered this option. It's not work in this case:

```php
final class SimpleModel implements ValidatedModelInterface
{
    use ValidatedModelTrait;

    public function __construct(
        #[PreValidate(new Required())]
        #[Length(min: 3, skipOnError: true)]
        public string $a = '.',
        public string $b = '.',
        public string $c = '.',
    ) {
    }
}

$service->create(SimpleModel::class, data: []);
```

`skipOnError` options work incorrect, because validator not know about errors on prevalidation level.

Reply:
OK. For `skipOnError`, indeed, order matters.